lib/ansible/modules/cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -590,10 +599,13 @@ def ensure_route_table_present(connection, module):
         changed = changed or result['changed']
 
     if not tags_valid and tags is not None:
-        result = ensure_tags(connection, route_table.id, tags,
-                             add_only=True, check_mode=module.check_mode)
-        route_table.tags = result['tags']
-        changed = changed or result['changed']
+        if module.check_mode and not route_table:
```

Comment:
For clarity, this should probably be

```
if module.check_mode:
    if not route_table:
        changed = changed or True
else:
    result = ensure_tags(...)
```
to make it absolutely clear that you can't hit the `else` branch with `route_table` being `None`